### PR TITLE
records: fix revisions revert

### PR DESCRIPTION
**Pull request by pazembrz**
*INSPIR-3486

### CODE REVIEW

File: backend/inspirehep/records/api/base.py [hunk 1]
```diff
@@ -451,3 +454,49 @@ def serialize_for_es(self):
...
+
+
+class InspireRevisionsIterator(RevisionsIterator):
+    def __getitem__(self, revision_id):
+        """Get a specific revision.
+
+        Revision id is always smaller by 1 from version_id.
+        Replaces RevisionsIterator from Invenio until they merge proper patch
+        """
+        if revision_id < 0:
```
Comment (drjova) [inline backend/inspirehep/records/api/base.py:491]: shouldn't this be `< 1`?

Comment (pazembrz) [reply backend/inspirehep/records/api/base.py:491]: no, as usually revision 0 is correct. But in Inspire it's impossible to have revision 0 as we are doing increment by 2 when we create record.

**Comment by pazembrz** [comment]
I have to change this PR as Invenio already merged my change, so I can remove like half of this PR. Don't approve yet please.

**Comment by pazembrz** [comment]
Ok, After discussion with Harrun I won't update invenio-records and we will stay with fix hard-coded in our code ad I would have to update invenio-records from 1.0.2 to 1.3.2

**Comment by drjova** [comment]
@pazembrz I'd still bump it if it doesn't require to bump everything.

File: backend/tests/integration-async/records/api/test_base.py [hunk 2]
```diff
@@ -140,3 +138,57 @@ def test_get_modified_references_returns_no_references_when_non_impacting_metada
     db.session.commit()
 
     assert citing_record.get_modified_references() == []
+
+
+def test_revert_revision_works_correctly_and_runs_update(inspire_app):
```
Comment (drjova) [inline backend/tests/integration-async/records/api/test_base.py:143]: Shall we test also that the data is correct after the revert?

Comment (pazembrz) [reply backend/tests/integration-async/records/api/test_base.py:143]: Check is already there (check line 194)

**Pull request merged**